projects/step-frontend/src/lib/modules/timeseries/pages/dashboards-list/dashboards-list-page.component.html

```diff
@@ -0,0 +1,115 @@
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="description" stepSearchCol>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header>Description</th>
+    <td mat-cell *matCellDef="let element">{{ element.description }}</td>
+  </ng-container>
+  <ng-container matColumnDef="dashlets">
+    <th mat-header-cell *matHeaderCellDef>Dashlets</th>
+    <td mat-cell *matCellDef="let element">
+      <step-icon
+        class="dashlet-icon"
+        [matTooltip]="dashlet.name"
+        *ngFor="let dashlet of element.dashlets"
+        name="trello"
+        style="margin-right: 4px"
+      ></step-icon>
+    </td>
+  </ng-container>
+  <ng-container matColumnDef="actions">
+    <th mat-header-cell *matHeaderCellDef>Actions</th>
+    <td mat-cell *matCellDef="let element" class="cell-actions">
+      <step-btn-group>
+        <button
+          mat-icon-button
+          matTooltip="View dashboard"
+          [disabled]="!('dashboard-write' | hasRight | async)"
+          (click)="navigateToDashboard(element)"
+        >
+          <step-icon name="eye"></step-icon>
+        </button>
+        <ng-container *ngIf="!element.metadata?.isLegacy">
+          <button
+            mat-icon-button
+            matTooltip="Edit dashboard"
+            [disabled]="!('dashboard-write' | hasRight | async)"
+            (click)="navigateToDashboard(element, true)"
+          >
+            <step-icon name="edit"></step-icon>
+          </button>
+          <button
+            mat-icon-button
+            matTooltip="Delete dashboard"
+            [disabled]="!('dashboard-delete' | hasRight | async)"
+            (click)="delete(element)"
+          >
+            <step-icon name="trash-2"></step-icon>
+          </button>
+        </ng-container>
+      </step-btn-group>
+    </td>
+  </ng-container>
+</step-table>
+
+<ng-template #createTemplate>
+  <step-modal-window title="New Dashboard">
+    <div step-modal-window-body>
+      <step-form-field>
+        <step-label>Name</step-label>
+        <input [(ngModel)]="createDashboardModel.name" />
+      </step-form-field>
+      <step-form-field>
+        <step-label>Description</step-label>
+        <textarea [rows]="4" [(ngModel)]="createDashboardModel.description"></textarea>
+      </step-form-field>
+    </div>
+    <div step-modal-window-buttons>
+      <button type="button" mat-stroked-button mat-dialog-close>Cancel</button>
+      <button
+        type="button"
+        [disabled]="!createDashboardModel.name"
+        mat-flat-button
+        color="primary"
+        (click)="saveNewDashboard()"
+      >
+        Save
+      </button>
+      <button
+        type="button"
+        [disabled]="!createDashboardModel.name"
+        mat-flat-button
+        color="primary"
+        (click)="saveNewDashboard(true)"
+      >
+        Save and edit
+      </button>
+    </div>
+  </step-modal-window>
+</ng-template>
```

Comment:
I would prefer to move it in a separate component

Reply:
@dvladir I would leave it for now just for the sake of tight timeline we're in, if you're ok with it